lib/smoke_test.rb

```diff
@@ -0,0 +1,61 @@
+require 'net/http'
+
+class SmokeTest
+  def initialize(urls)
+    @@urls = urls
+    secrets = Rails.application.secrets
+    @@api_token = secrets.smoke_test[:api_token] || ""
+    @@slack_uri = secrets.smoke_test[:slack_uri]
+    @@slack_token = secrets.smoke_test[:slack_token]
+    @@slack_room = secrets.smoke_test[:slack_room]
+    @@slack_username = secrets.smoke_test[:slack_username]
+    @@slack_emoji = secrets.smoke_test[:slack_emoji]
+  end
+
+  def self.test_endpoints
+    message = ""
+
+    @@urls.each do |url|
+      if /api/.match(url)
+        curl_result = `curl -i -s -w "%{http_code}" #{url} -H "X-Authentication-Token:#{@@api_token}" -o /dev/null`
```

Comment:
This bit for the API is probably too specific.
It is true that we mostly use `X-Authentication-Token` as a field to store the token in the http request, but not everybody uses the same approach.

One way to generalise this a little bit more, would be to ask the user to customise that http field; so perhaps adding an extra variable in the secrets (and .env file) could be ok at this point.
In the future you can consider a better way to allow the user to configure the gem as she/he likes.

Reply:
Sounds like a great idea thanks @Levia!

Reply:
@Levia I was thinking I could rename the `@api_token` to something like `@x_authentication_token` and explain in the docs about how this is used. And if blank, I can just not submit that token to the server.